lib/api/scenarios.ts

```diff
@@ -0,0 +1,44 @@
+import { ShredderStorage } from './storage';
+import { Scenario } from './models/scenario';
+import { Rule } from './models/rule';
+
+export class Scenarios {
+
+  public scenarios = [];
+  public currentScenario;
+
+  constructor() {
+    this.loadFromStorage();
+  }
+
+  loadFromStorage() {
+    const fromStorage = ShredderStorage.getSerialized();
+
+    fromStorage.scenarios.forEach((scenario) => {
+      const { id, name } = scenario;
+      const rules = [];
+
+      scenario.rules.forEach((rule) => rules.push(new Rule(rule)));
```

Comment:
Maybe just pass the rules into the scenario model and that will create the rule models?